localstack/services/sns/resource_providers/aws_sns_topic.py

```diff
@@ -0,0 +1,185 @@
+    KmsMasterKeyId: Optional[str]
+    SignatureVersion: Optional[str]
+    Subscription: Optional[list[Subscription]]
+    Tags: Optional[list[Tag]]
+    TopicArn: Optional[str]
+    TopicName: Optional[str]
+    TracingConfig: Optional[str]
+
+
+class Subscription(TypedDict):
+    Endpoint: Optional[str]
+    Protocol: Optional[str]
+
+
+class Tag(TypedDict):
+    Key: Optional[str]
+    Value: Optional[str]
+
+
+REPEATED_INVOCATION = "repeated_invocation"
+
+
+class SNSTopicProvider(ResourceProvider[SNSTopicProperties]):
+    TYPE = "AWS::SNS::Topic"  # Autogenerated. Don't change
+    SCHEMA = util.get_schema_path(Path(__file__))  # Autogenerated. Don't change
+
+    def create(
+        self,
+        request: ResourceRequest[SNSTopicProperties],
+    ) -> ProgressEvent[SNSTopicProperties]:
+        """
+        Create a new resource.
+
+        Primary identifier fields:
+          - /properties/TopicArn
+
+
+
+        Create-only properties:
+          - /properties/TopicName
+          - /properties/FifoTopic
+
+        Read-only properties:
+          - /properties/TopicArn
+
+        IAM permissions required:
+          - sns:CreateTopic
+          - sns:TagResource
+          - sns:Subscribe
+          - sns:GetTopicAttributes
+          - sns:PutDataProtectionPolicy
+
+        """
+        model = request.desired_state
+        sns = request.aws_client_factory.sns
+        # TODO: validations and iam checks
+
+        attributes = {k: v for k, v in model.items() if v is not None if k != "TopicName"}
+
+        # following attributes need to be str instead of bool for boto to work
+        if attributes.get("FifoTopic") is not None:
+            attributes["FifoTopic"] = str(attributes.get("FifoTopic"))
+
+        if attributes.get("ContentBasedDeduplication") is not None:
+            attributes["ContentBasedDeduplication"] = str(
+                attributes.get("ContentBasedDeduplication")
+            )
+
+        subscriptions = []
+        if attributes.get("Subscription") is not None:
+            subscriptions = attributes["Subscription"]
+            del attributes["Subscription"]
+
+        tags = []
+        if attributes.get("Tags") is not None:
+            subscriptions = attributes["Tags"]
+            del attributes["Tags"]
+
+        # in case cloudformation didn't provide topic name
+        if model.get("TopicName") is None:
+            model["TopicName"] = f"topic-{short_uid()}"
+
+        create_sns_response = sns.create_topic(Name=model["TopicName"], Attributes=attributes)
+        request.custom_context[REPEATED_INVOCATION] = True
+        model["TopicArn"] = create_sns_response["TopicArn"]
+
+        # now we add subscriptions if they exists
+        # TODO do we need to keep arns for unsubscribing before deleting topic
+        for subscription in subscriptions:
+            sns.subscribe(
+                TopicArn=model["TopicArn"],
+                Protocol=subscription["Protocol"],
```

Comment:
Seems this is optional (latest CI run fails here with a `KeyError`) and might either need to be set to a default or dropped from the subscribe call if it doesn't exist. 

```
    event = self.execute_action(resource_provider, payload)
  File "/opt/code/localstack/localstack/services/cloudformation/resource_provider.py", line 712, in execute_action
    return resource_provider.create(request)
  File "/opt/code/localstack/localstack/services/sns/resource_providers/aws_sns_topic.py", line 114, in create
    Protocol=subscription["Protocol"],
KeyError: 'Protocol'
```

Reply:
I wasn't sure what a default could be so, I've added the code that drops subscription in case there Protocol field is missing.
Maybe I could add a warning as well?